src/main/java/org/sqlite/core/NativeDB.c

```diff
@@ -474,11 +490,23 @@ JNIEXPORT jint JNICALL JNI_OnLoad(JavaVM *vm, void *reserved)
     phandleclass = (*env)->FindClass(env, "org/sqlite/ProgressHandler");
     if(!phandleclass) return JNI_ERR;
     phandleclass = (*env)->NewWeakGlobalRef(env, phandleclass);
+    phandle_mth_progress = (*env)->GetMethodID(env, phandleclass, "progress", "()I");
 
-    jclass exclass = (*env)->FindClass(env, "java/lang/Throwable");
+    bhandleclass = (*env)->FindClass(env, "org/sqlite/BusyHandler");
+    if(!bhandleclass) return JNI_ERR;
+    bhandleclass = (*env)->NewWeakGlobalRef(env, bhandleclass);
+    bhandle_mth_callback = (*env)->GetMethodID(env, bhandleclass, "callback", "(I)I");
+
+    exclass = (*env)->FindClass(env, "java/lang/Throwable");
+    if(!exclass) return JNI_ERR;
+    exclass = (*env)->NewWeakGlobalRef(env, exclass);
     exp_msg = (*env)->GetMethodID(
             env, exclass, "toString", "()Ljava/lang/String;");
 
+    bool_array_class = (*env)->FindClass(env, "[Z");
+    if(!bool_array_class) return JNI_ERR;
+    bool_array_class = (*env)->NewWeakGlobalRef(env, bool_array_class);
```

Comment:
The jni-config entry for `boolean[]` was missing, which caused the `java.sql.SQLException: Out of memory` mentioned in the surefire exclusions.

Reply:
Do we need to maintain a global weak reference to this, can we just make sure that the returned class is not null which implies it is accessible from JNI?

Reply:
Do you mean not storing the class ref and doing the `FindClass` call again later when needed? While that would work, I think it would be a bit pointless. The lookup itself is "slow" so it makes sense to cache it if the lookup happens anyway in the `JNI_OnLoad`. It would also break with my suggestion to keep JNI lookups confined in the `JNI_OnLoad` function.